# Type: fix
# Subtype: I8
# English Summary: Correctly parse memory mapping settings
# Chinese Summary: 正确解析内存映射设置
# Entry Number: 17

# === Git Diff Content ===

diff --git a/enterprise/enterprise-performance-tests/src/main/java/org/neo4j/perftest/enterprise/ccheck/DataGenerator.java b/enterprise/enterprise-performance-tests/src/main/java/org/neo4j/perftest/enterprise/ccheck/DataGenerator.java
index 3417aff..5c05cf7 100644
--- a/enterprise/enterprise-performance-tests/src/main/java/org/neo4j/perftest/enterprise/ccheck/DataGenerator.java
+++ b/enterprise/enterprise-performance-tests/src/main/java/org/neo4j/perftest/enterprise/ccheck/DataGenerator.java
@@ -349,7 +349,8 @@ public class DataGenerator
 
     private static Map<String, String> batchInserterConfig( Configuration configuration )
     {
-        Long mappedMemory = Long.parseLong( configuration.get( all_stores_total_mapped_memory_size ) );
+        Long mappedMemory = GraphDatabaseSettings.all_stores_total_mapped_memory_size
+                .valueOf( configuration.get( all_stores_total_mapped_memory_size ), null );
 
         Map<String, String> config = new HashMap<String, String>();
         config.put( "use_memory_mapped_buffers", "true" );
